crates/vault-registry/src/lib.rs

```diff
@@ -861,7 +905,7 @@ impl<T: Config> Pallet<T> {
         };
 
         let is_below_threshold =
-            Pallet::<T>::is_collateral_below_secure_threshold(&new_collateral, &vault.backed_tokens()?)?;
+            Pallet::<T>::is_collateral_below_vault_secure_threshold(&new_collateral, &vault.backed_tokens()?, &vault)?;
```

Comment:
Would be good to remove the `is_collateral_below_secure_threshold` function since we don't use it anymore. If we leave the function around we risk using it in the future when we should instead use the new one